classes/navigationMenu/NavigationMenuItemDAO.inc.php

```diff
@@ -517,6 +517,35 @@ function unCacheRelatedNavigationMenus($id) {
 			}
 		}
 	}
+
+	/**
+	 * Port static page as a Custom NMI
+	 * @param StaticPage $staticPage
+	 * @return int The id of the inserted NMI. Null if non is inserted
+	 */
+	function portStaticPage($staticPage) {
+		$path = $staticPage->getPath();
+		$contextId = $staticPage->getContextId();
+
+		$existingNMIWithPath = $this->getByPath($contextId, $path);
+
+		$retNavigationMenuItemId = null;
+
+		if (!isset($existingNMIWithPath)) {
```

Comment:
In the case there *is* a conflicting NMI, I'd suggest using `error_log` to log a warning. (See `classes/install/Upgrade.inc.php` for other examples of this.)

Reply:
@asmecher I added the following message
`WARNING: The StaticPage "tetst" uses a path (test) that conflicts with an existing Custom Navigation Menu Item path. Skipping this StaticPage.`